### ISSUE TITLE
Calling VoiceClient::playFile second time on the same client produces no actual sound

### ISSUE THREAD

**Issue by rederrik** [teamreflex/DiscordPHP#444]
**Environment**
- PHP Version:
    - PHP 7.4.12 (cli) (built: Oct 31 2020 17:03:53) ( NTS )
- DiscordPHP Version:
    - dev-master (064baaa2f6e301de3fb1f37d5defaf09e7d436e3), 5.0.12

**Describe the bug**
After calling `VoiceClient::playFile` second (and further) time using the existing VoiceClient instance you can hear no actual sound coming from the bot, although the outlie still flashes green (speaking). No errors are produced too.

**To Reproduce**
- join voice channel
- play sound
- play sound again

In the following example, sending `playtest` command twice produces such behavior. 
```php
//dev-master example
$discord->registerCommand('playtest', function (Message $message) use ($discord) {
        if ($voice = $discord->getVoiceClient($message->channel->guild_id)) { //Try to get an active voice client
            $voice->playFile(__DIR__.'/sounds/murloc.wav');
            return null;
        }

        $voiceStates = $message->channel->guild->voice_states;
        $userId = $message->author->id;

        foreach ($voiceStates as $voiceState) { //Find a voice channel the user is in
            if ($voiceState->user_id === $userId) {
                $channel = $discord->getChannel($voiceState->channel_id);
            }
        }

        if (empty($channel)) { //Channel not found
            return "Can't join your current voice channel";
        }

        $discord->joinVoiceChannel($channel)->then(  //Join a channel if we dont have an active voice client
            function (VoiceClient $voice) {
                $voice->playFile(__DIR__.'/sounds/murloc.wav');
            }
        );

        return null;
    }
);
```

**Expected behavior**
`VoiceClient::playFile` correctly producing sound multiple times before closing the client.

**Additional context**
One could add `->then([$voice, 'close'])` to `playSound` call as a workaround - this would close the client, dropping the bot from the channel, but having to join and leave each time is annoying.

**Comment by davidcole1340** [comment]
I think you are trying to play two things at once? Does the error still happen if the song plays out then you try and play it again?

Also, it's better to use `->done()` instead of `->then()` at the end of your promise chains as exceptions might be supressed otherwise:

```php
$promise->then(function () {
    // do something
})->then(function () {
    // do something else
})->done();
```

**Comment by rederrik** [comment]
It happens after you wait for a while and call it again, as long as you use the same client without leaving the channel.
In my example, you are supposed to call this command twice: first the bot join a channel and plays a sound, second time it just plays a sound.

But sure enough, calling it once using the following code has the same issue:
```php
$discord->joinVoiceChannel($channel)->done(  //Join a channel if we dont have an active voice client
    function (VoiceClient $voice) {
        $voice->playFile(__DIR__.'/sounds/murloc.wav')->done(function () use ($voice) {
            $voice->playFile(__DIR__.'/sounds/murloc.wav');
        });
    }
);
```

**Comment by davidcole1340** [comment]
Could you try this:
```php
$discord->joinVoiceChannel($channel)->then(function (VoiceClient $voice) {
    return $voice->playFile(__DIR__.'/sounds/murloc.wav');
})->then(function () {
    return $voice->playFile(__DIR__.'/sounds/murloc.wav');
})->done();
```

**Comment by rederrik** [comment]
Neither of these work (line 3), $voice is either `null` or `bool` in the second lambda:
```php
})->then(function () {
```
```php
})->then(function () use($voice) {
```
```php
})->then(function ($voice) {
```

**Comment by davidcole1340** [comment]
Oops:
```php
$discord->joinVoiceChannel($channel)->done(function (VoiceClient $voice) {
    $voice->playFile(__DIR__.'/sounds/murloc.wav')->then(function () use ($voice) {
        return $voice->playFile(__DIR__.'/sounds/murloc.wav');
    })->done();
});
```

**Comment by rederrik** [comment]
This works without throwing any errors, but still the same issue - no sound for the second time it plays.

**Comment by davidcole1340** [comment]
Exact same code works for me - are you sure there isn't something wrong with the file? Maybe try this to see if the songs finish:

```php
->done(function () {
    echo 'Done'.PHP_EOL;
});
```

**Comment by rederrik** [comment]
Yeah I see "Done" output as expected.

Could you please share the full code that works for you?

Also, how long is the audio file you tested? 
I tried this with 2 different files and I feel like there is some weird silence period when the next file starts to play - I could hear maybe last 40% of the second test file which was also longer than the first one.

**Comment by davidcole1340** [comment]
I tried with two files - one about 7 mins and another about 30s, my code is almost identical:

```php
$channel = $discord->getChannel('channel_id');
$discord->joinVoiceChannel($channel)->done(function (VoiceClient $vc) {
    $vc->playFile('/sandman.mp3')
    ->then(fn () => $vc->playFile('/sandman.mp3'))
    ->done(fn () => var_dump('Done'));
});
```

Could you let me know some things:
- OS
- PHP Version - `php -v`
- DiscordPHP Version - `composer show team-reflex/discord-php`

**Comment by rederrik** [comment]
Oh well.. I didn't have the patience to test with larger files, so I used samples 1-3 seconds long.
Now when I did try 2 minute long song I can see the issue more clear.

So here is what I did:
- have 2 commands like in original example, `cmd1` with 2 second long .wav file and `cmd2` with 2 minute long .mp3 song
- call `cmd1` once - playback ok
- call `cmd1` again after a while - hear nothing
- call `cmd2` - hear the song playback _without a couple of first seconds of it_
- call `cmd2` again after the playback finished - hear the song playback _including missing first seconds of it_
- can cycle the steps and get the same results now

So basically it looks like some weird issue with short files to me.

To be even more specific, try using these free samples: 
- [this one](http://soundbible.com/2098-Grim-Reaper-Your-Soul.html) is short enough to cause this issue
- [this one](http://soundbible.com/2127-Dragon-Fire-Breath-and-Roar.html) is long enough to don't cause this issue

I'm running it on Ubuntu 16.04.7, PHP 7.4.12 (cli) ( NTS ), DiscordPHP dev-master (4e3a01a).

**Comment by davidcole1340** [comment]
Sweet I can reproduce now, I'll let you know when I can fix it. Cheers

**Comment by davidcole1340** [comment]
Seems to be quite a tricky bug. Unfortunately I don't have the time to invest into this at the moment sorry. Possibly pad the end of your song with silence as a temporary work around?

I'm fairly sure this could be fixed by sending a few packets of silence before ending transmission.

**Comment by rederrik** [comment]
It sure is!
Actually, I tried creating a 4 seconds silence.wav file, that would be played after each "normal" playback, but weirdly enough that didn't do anything to mitigate this issue. Maybe this is some kind of buffer issue and actual size matters, not length?

Also, for some reason the sound is all choppy now using the latest version. Does it sound good for you?

**Comment by key2peace** [comment]
do you folks test if the client is still playing a voice file and if yes, either stop it before making it play the next file?

**Comment by rederrik** [comment]
> do you folks test if the client is still playing a voice file and if yes, either stop it before making it play the next file?

Well there are code samples up there in this thread, as well as test sequence.. or you mean in general?

**Comment by key2peace** [comment]
https://discord.com/developers/docs/topics/opcodes-and-status-codes#voice

I am talking about opcode 5 here, check if the client is still speaking (playing music), if yes, stop the playing before telling it to play again

**Comment by key2peace** [comment]
ok, I got it to playFile a second time in the channel by issueing a $vc->stop() before starting playFile again, but the result of the audio of the second time is awfull, here is the rough code (skipping the join etc stuff, you get the idea behind it:
```
if ($vc = $discord->getVoiceClient($message->guild_id))
{
	if ($vc->isSpeaking($discord->id)) {
		$vc->stop()->then(function () use ($vc) {$vc->playFile('sample.mp3');});
	}
	else
		$vc->playFile('sample.mp3');
	
}
```
